crates/bitwarden/src/auth/password.rs

```diff
@@ -87,6 +87,7 @@ mod tests {
                 iterations: NonZeroU32::new(100_000).unwrap(),
             },
             client_id: "1".to_string(),
+            refresh_token: None,
```

Comment:
I'm a bit hesitant to couple refresh_token to login method. While it's true certain login methods get a refresh token while others don't, there is no direct relation to users.

Reply:
It's not so much which methods return a refresh token and which don't, but wether we actually use the refresh_token for that method when renewing tokens.

At the moment we have the following renewal patterns:
- User with username: We use client_id and refresh_token
- User with apikey: We use client_id and client_secret
- Service account with access token: We use the access token's id and client_secret

So even if the other methods return a refresh_token, it would only be used in the case of user with username.

Reply:
In that case what do you think about changing the flow to always use refresh_token if set and only falling back to the generic renew if it doesn't exist?

Reply:
Hmm, that can work, but we would also need to pull the client_id from each login_method and into the client, as we need it for the token refresh request.